backends/cuda/ceed-cuda-vec.c

```diff
@@ -61,9 +61,10 @@ static int CeedVectorSetArrayHost_Cuda(const CeedVector vec,
   case CEED_COPY_VALUES: ;
     CeedInt length;
     ierr = CeedVectorGetLength(vec, &length); CeedChk(ierr);
-    ierr = CeedMalloc(length, &data->h_array_allocated); CeedChk(ierr);
+    if (!data->h_array_allocated) {
```

Comment:
@YohannDudouit I added this because I found in the OCCA backend that SetArray to set a memory location followed by SetArray to copy data was not copying data into the previously set memory location. Also added below for the device memory.

Reply:
I'm not sure I completely agree with this, if you do `SetArray(vec, CEED_USE_POINTER, &array)` followed by `SetArray(vec, CEED_COPY_VALUES, &array)` it would replace `h_array` by a newly allocated vector, but if you do `SetArray(vec, CEED_OWN_POINTER, &array)` followed by `SetArray(vec, CEED_COPY_VALUES, &array)` then it copies into `h_array`, is it the expected behaviour?

Reply:
Are you sure it's not:
```c++
if (!data->h_array)
```
?